test/target_test.py

```diff
@@ -266,3 +268,68 @@ def test_rename_dont_move_on_fs(self):
         self.assertFalse(t.exists())
         self.assertRaises(luigi.target.FileAlreadyExists,
                           lambda: fs.rename_dont_move(t.path, t.path+"-yay"))
+
+
+class TemporaryPathTest(unittest.TestCase):
+    def setUp(self):
+        super(TemporaryPathTest, self).setUp()
+        self.fs = Mock()
+
+        class MyFileSystemTarget(luigi.target.FileSystemTarget):
+            open = None  # Must be implemented due to abc stuff
+            fs = self.fs
+
+        self.target_cls = MyFileSystemTarget
+
+    def test_temporary_path_files(self):
+        target_outer = self.target_cls('/tmp/notreal.xls')
+        target_inner = self.target_cls('/tmp/blah.txt')
+
+        class MyException(Exception):
+            pass
+
+        orig_ex = MyException()
+        try:
+            with target_outer.temporary_path() as tmp_path_outer:
+                assert 'notreal' in tmp_path_outer
+                with target_inner.temporary_path() as tmp_path_inner:
+                    assert 'blah' in tmp_path_inner
+                    with target_inner.temporary_path() as tmp_path_inner_2:
+                        assert tmp_path_inner != tmp_path_inner_2
+                    self.fs.rename_dont_move.assert_called_once_with(tmp_path_inner_2, target_inner.path)
+                self.fs.rename_dont_move.assert_called_with(tmp_path_inner, target_inner.path)
+                self.fs.rename_dont_move.call_count == 2
+                raise orig_ex
+        except MyException as ex:
+            self.fs.rename_dont_move.call_count == 2
+            assert ex is orig_ex
+        else:
+            assert False
+
+    def test_temporary_path_directory(self):
+        target_slash = self.target_cls('/tmp/dir/')
+        target_noslash = self.target_cls('/tmp/dir')
+
+        with target_slash.temporary_path() as tmp_path:
+            assert re.match(r'/tmp/dir-luigi-tmp-\d{10}/', tmp_path)
+        self.fs.rename_dont_move.assert_called_once_with(tmp_path, target_slash.path)
+
+        with target_noslash.temporary_path() as tmp_path:
+            assert re.match(r'/tmp/dir-luigi-tmp-\d{10}', tmp_path)
+        self.fs.rename_dont_move.assert_called_with(tmp_path, target_noslash.path)
+
+    def test_windowsish_dir(self):
+        target = self.target_cls(r'''C:\my\folder''' + "\\")
+        pattern = r'''C:\\my\\folder-luigi-tmp-\d{10}''' + r"\\"
+
+        with target.temporary_path() as tmp_path:
+            assert re.match(pattern, tmp_path)
+        self.fs.rename_dont_move.assert_called_once_with(tmp_path, target.path)
+
```

Comment:
To save you the trouble of reading the Travis report, your only issue was a flake8 complaint. You included too may `\n` here.